utilities/fault_injection_fs.cc

```diff
@@ -803,7 +803,7 @@ IOStatus FaultInjectionTestFS::GetFileSize(const std::string& f,
     // Need to report flushed size, not synced size
     MutexLock l(&mutex_);
     auto it = db_file_state_.find(f);
-    if (it != db_file_state_.end()) {
+    if (it != db_file_state_.end() && it->second.pos_ > 0) {
```

Comment:
Without a comment, it's not clear why this check should be here.

The core problem seems to be that we aren't simulating unsynced data loss with direct_io, and because of that, we aren't setting pos_. Why should this code be quietly entangled with that deficit of other code (rather than fixing the other code)?

It seems like it would be less code to treat use_direct_io the same way as !use_direct_io and either not use direct_io in target_ or create an aligned buffer when we do append to target_.

Reply:
> The core problem seems to be that we aren't simulating unsynced data loss with direct_io, and because of that, we aren't setting pos_. Why should this code be quietly entangled with that deficit of other code (rather than fixing the other code)?

This is fair.

Reply:
> It seems like it would be less code to treat use_direct_io the same way as !use_direct_io and either not use direct_io in target_ or create an aligned buffer when we do append to target_.

Or easy to update pos_ with use_direct_io.

Let me think about this.

Reply:
I think the direct IO is meant for skipping the page cache and therefore we don't simulate un-synced data loss like loss in page cache for direct IO. `update pos_ with use_direct_io.` seems doable.

Reply:
After more discussion, we agreed upon direct IO skipping page cache doesn't necessarily there is no chance for data loss upon power loss. So we still want to simulate data loss for direct IO write in the future. 

Fixed by "update pos_ with use_direct_io" for now.